### DISCUSSION TITLE
empty field while calling API

### DISCUSSION THREAD

**Discussion by Jefflai0315** [Budibase/budibase#4520] [Help]
![image](https://user-images.githubusercontent.com/79297365/154204029-3df93840-640f-4999-90db-6e0474e759ad.png)

![image](https://user-images.githubusercontent.com/79297365/154204223-ff42105c-ab77-4268-a077-9adc292750cd.png)

Hi, could you advise what have I not done on the Frontend? I am able to call the API in the Data but not in the Design ( when one of my fields is left empty)

**Comment by mjashanks** [comment]
It's hard to say without seeing your app. 

Are you passing bindings into your query, via the cog icon...

![image](https://user-images.githubusercontent.com/3524181/154240635-5d179163-b58d-4643-8649-2bb9796ea15b.png)

Also, how are you using the bindings in your query? Maybe you need to give it an actual value. The issue could be because that binding is being passed `undefined` rather than a blank string.

Maybe when you setup your bindings in the Design section, you need to do something like

```javascript
const parametersInput = $("Form.Fields.parameters");
if (parametersInput) return parametersInput;
else return parametersInput; 
```

**Comment by Jefflai0315** [comment]
yes i am using binding in query and I left it without a default value. 

the query sometimes does not require any input, so putting an actual value wont help. I tried your methods but it is still the same 
![image](https://user-images.githubusercontent.com/79297365/154265804-77191a4d-4c8e-4030-9f56-7f2bff456bf4.png)

also i realized that i am not able to change the function any more? help
![image](https://user-images.githubusercontent.com/79297365/154266007-784ce283-9ac3-4dbb-9247-29d00a882d43.png)

**Reply by Jefflai0315** [reply]
Hi @mjashanks, Using Budibase has been great so far! appreciate it 


I just encountered this issue that I am not able to modify my javascript/handlebar once I created them. 
another question is that it shows I have exceeded the quota of 4000 rows, which hinders me to save any new rows..
Also Could you please advise?

**Reply by mjashanks** [reply]
Hey @Jefflai0315 - this is an issue that we are looking into at the moment. Going to get this fixed asap...

**Reply by Jefflai0315** [reply]
thank you, really appreciate the team behind budibase.
